tests/integration/test_charm.py

```diff
@@ -40,3 +45,34 @@ async def test_relating_to_loki(ops_test):
     await ops_test.model.deploy("loki-k8s", channel="edge", application_name="loki")
     await ops_test.model.add_relation("loki", "agent:logging-consumer")
     await ops_test.model.wait_for_idle(apps=["loki", "agent"], status="active", timeout=1000)
+
+
+async def test_relating_to_grafana(ops_test):
+    await ops_test.model.deploy("grafana-k8s", channel="edge", application_name="grafana")
+    await ops_test.model.add_relation("grafana", "agent:grafana-dashboard")
+    await ops_test.model.wait_for_idle(apps=["agent", "grafana"], status="active", timeout=1000)
+    dashboards = await get_grafana_dashboards(ops_test, "grafana", 0)
+    assert dashboards[0]["title"] == "Grafana Agent"
+
+
+async def test_relating_to_prometheus(ops_test):
+    await ops_test.model.deploy(
+        "prometheus-k8s", channel="edge", application_name="prometheus", trust=True
+    )
+    await ops_test.model.add_relation("prometheus", "agent:self-metrics-endpoint")
+    await ops_test.model.wait_for_idle(apps=["agent", "prometheus"], status="active", timeout=1000)
+    alert_rules_names = [
+        "GrafanaAgentRequestErrors",
+        "GrafanaAgentRequestLatency",
+        "GrafanaAgentUnavailable",
+    ]
+    alert_rules = await get_prometheus_rules(ops_test, "prometheus", 0)
+    assert len(alert_rules) == 3
```

Comment:
Seems a bit brittle. Could something like `ops_tests.juju("exec", ...)` the analogous of
```bash
juju exec --unit prom/0 -- ls /var/lib/juju/agents/unit-prom-0/charm/src/prometheus_alert_rules
```
help here?

Reply:
What `get_prometheus_rules` does is to query the `/api/v1/rules` prometheus endpoint to get the `rules`.... Why do you think is brittle?